### PR TITLE
GC Code: J9MMCONSTANT_IMPLICIT_GC_COMPLETE_CONCURRENT

### PR DESCRIPTION
**Pull request by RSalman**
Introduced GC code `J9MMCONSTANT_IMPLICIT_GC_COMPLETE_CONCURRENT` which
is used to signal that a concurrent phase is being forced completed -
used through a call to `localGarbageCollect(MM_EnvironmentBase *env,
uint32_t gcCode)`

Required for https://github.com/eclipse/openj9/pull/7161

Signed-off-by: Salman Rana <salman.rana@ibm.com>

### CODE REVIEW

File: gc/base/MemorySpace.hpp [hunk 1]
```diff
@@ -155,7 +155,7 @@ class MM_MemorySpace : public MM_BaseVirtual
 	MMINLINE MM_MemorySubSpace *getMemorySubSpaceList() { return _memorySubSpaceList; }
 
 	void systemGarbageCollect(MM_EnvironmentBase *env, uint32_t gcCode);
-	void localGarbageCollect(MM_EnvironmentBase *env);
```
Comment (RSalman) [inline gc/base/MemorySpace.hpp]: Need default value due to cross dependent changes - default value will be removed once both OMR and OpenJ9 are merged.

File: gc/base/MemorySpace.hpp [hunk 2]
```diff
@@ -155,7 +155,7 @@ class MM_MemorySpace : public MM_BaseVirtual
...
+	void localGarbageCollect(MM_EnvironmentBase *env, uint32_t gcCode = J9MMCONSTANT_IMPLICIT_GC_DEFAULT);
```
Comment (RSalman) [inline gc/base/MemorySpace.hpp]: Need default value due to cross dependent changes - default value will be removed once both OMR and OpenJ9 are merged.

**Comment by rwy0717** [comment]
LGTM, but I'll wait for @RSalman to remove the WIP tag.

**Comment by RSalman** [comment]
Removed tag

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by RSalman** [comment]
Checks passed, can I please have this merged if there are no more reviews

**Pull request merged**